### DISCUSSION TITLE
TuYa ZY-M100-24G Human presence sensor does not save values and spams the network

### DISCUSSION THREAD

**Discussion by SonTypda** [Koenkk/zigbee2mqtt#21180] [Devices]
Hi everyone,
I have been using two  TuYa ZY-M100-24G Human presence sensors for a few days now.

I was wondering why both sensors changed the status from present to absent and vice versa every minute, even though nobody was present.
I then saw that the "Presence sensitivity" value was set to the maximum value (10) for both sensors.
As soon as I lower the value again, it doesn't take long and it is automatically set to 10 again. 
I created an automation that immediately lowers the value as soon as it increases. This allows me to track how often this happens.

The whole thing also means that the Zigbee network is literally spammed, as the sensors constantly detect a presence at the highest sensitivity.

I use the latest version of Zigbee2MQTT.
According to the OTA, there is no update available for the sensors. 
Does anyone have the same problem and/or perhaps know a solution here?

Perhaps the presence sensitivity value could be set so that it can no longer be changed automatically? 

I would be very happy about help and answers :) 

![grafik](https://github.com/Koenkk/zigbee2mqtt/assets/151165250/58ce24ce-f5d9-45d9-aeea-fd0c8751c602)
Here you can see the log for one of those sensors. 
It literally spammed my network.

![Bildschirmfoto vom 2024-02-03 08-00-30](https://github.com/Koenkk/zigbee2mqtt/assets/151165250/8c2ca839-1efa-4626-aee6-5cc9788eacaf)
Here you can see how often (through my automation) the value of the presence sensibility was changed.
And that only within a few hours...

**Comment by AlfAlex** [comment]
There is also an open issue (#21282) but no solution yet :(

**Comment by AlfAlex** [comment]
Case #19448 shows a workaround

**Comment by jvdburgt** [comment]
There seems to be a solution available for the spamming problem, although it's a rather laborious one. See the last couple of posts [here](https://github.com/Koenkk/zigbee2mqtt/issues/19045). 

Thought it might be worthwhile mentioning here. Please update here if this works for you, and let's see if we can streamline this option perhaps.

**Reply by magicus** [reply]
Thanks for the link, I don't think I'd had found that otherwise! ❤️

**Comment by ESP4Ever** [comment]
Hi!

Also I noted that these sensors have a short update time and with a default keepalive interval of 60s they are flickering between "away - unavailable" states so I had to shorten the interval to 15s to fix it.

**Comment by asukahan** [comment]
I bought a small Tuya gateway, and removed my tuya sensors from Z2M, then binded them with the new tuya zigbee gateway. linked them with my z2m lights in Home assistant automation, my z2m now works like charm.

**Comment by Suschibomber** [comment]
Is there now a way that he doesn't always lose the last settings? And which tuya gateway did you buy? @asukahan
